encode.go

```diff
@@ -92,44 +115,49 @@ func (e *Encoder) ok(body interface{}) error {
 	return e.encode(body)
 }
 
-// Unauthorized writes a 401 status to the response header and returns the original error
-func (e *Encoder) Unauthorized(err error) error {
-	return e.statusCode(http.StatusUnauthorized, err)
+// BadRequest writes a http 400 status to the response header and returns the original error
+func (e *Encoder) BadRequest(err error) error {
+	return e.statusCode(http.StatusBadRequest, err)
 }
 
-// Forbidden writes a 403 status to the response header and returns the original error
-func (e *Encoder) Forbidden(err error) error {
-	return e.statusCode(http.StatusForbidden, err)
+// BadRequestWithMessage writes a bad request with a message body in the response
```

Comment:
```suggestion
// BadRequestWithMessage writes an http 400 status and a message body to the response and returns the original error
```

What I'm suggesting here is that we align all of the method descriptions (this would need to be done throughout the entire file).
Currently, the non-message version reads:

```
// BadRequest writes a http 400 status to the response header and returns the original error
func (e *Encoder) BadRequest(err error) error {
```

My suggested change above aims to follow the pattern defined here.  Additionally, we would want to change the Messagef version to:
`// BadRequestWithMessagef writes an http 400 status and a message body to the response and returns the original error`